### PR TITLE
Improve code quality and readd opentelmetry integration

### PR DESCRIPTION
**Pull request by MsRandom**

### CODE REVIEW

File: miner/src/main.rs [hunk 1]
```diff
@@ -102,13 +102,21 @@ impl Solver {
...
+                offset + read_len - read_len % (8 * 2),
+                read_len % (8 * 2),
+            );
         } else {
             unsafe {
                 self.solve_chunked::<4>(data_u8);
             }
 
-            self.solve(data, offset + read_len - read_len % (8 * 4), read_len % (8 * 4));
+            self.solve(
```
Comment (3xela) [inline miner/src/main.rs:115]: i like this

File: miner/src/main.rs [hunk 2]
```diff
@@ -102,13 +102,21 @@ impl Solver {
                 self.solve_chunked::<2>(data_u8);
             }
 
-            self.solve(data, offset + read_len - read_len % (8 * 2), read_len % (8 * 2));
+            self.solve(
```
Comment (3xela) [inline miner/src/main.rs:105]: this is good

**Review by 3xela — COMMENTED** [review]
not much to add, it looks good and you add the functionality for data collection

File: miner/src/main.rs (hunk 2 above)
Comment (MsRandom) [reply miner/src/main.rs:105]: it's autoformat lol

**Review by 3xela — APPROVED** [review]

**Pull request merged**